### ISSUE TITLE
Cookie Domain/Expiration are Undefined

### ISSUE THREAD

**Issue by topether21** [elysiajs/elysia#819]
### What version of Elysia is running?

1.1.12

### What platform is your computer?

Darwin 24.0.0 arm64 arm

### What steps can reproduce the bug?

Create a simple Elysia app

```
import { Elysia } from "elysia";
import { isHtml } from '@elysiajs/html'

const app = new Elysia()

app.get('/', () => '<h1>Hello World</h1>', {
  afterHandle({ response, set, cookie }) {
    console.log("afterHandle", cookie)
    cookie.my_cookie.expires = new Date(Date.now() + 1000 * 60 * 60 * 24)
    cookie.my_cookie.value = 'my_cookie_value'
    console.log("cookie value", cookie.my_cookie.value)
    console.log("cookie expires", cookie.my_cookie.expires)
    if (isHtml(response))
      set.headers['content-type'] = 'text/html; charset=utf8'
  }
})

app.listen(3000);

console.log(
  `🦊 Elysia is running at http://${app.server?.hostname}:${app.server?.port}`
);
```

```
1. Run the application.
2. Check the cookie handling logic.
3. Observe the output for the cookie expiration.
```

### What is the expected behavior?

The cookie expires should have a defined value indicating when the cookie will expire.

### What do you see instead?

I encountered an issue where the cookie expiration is returning as undefined, which is not the expected behavior.
The output shows cookie expires undefined, which suggests that the expiration date is not being set or retrieved correctly.

![image](https://github.com/user-attachments/assets/eda31137-733b-4f4d-9a08-7f0a2d07a090)

### Have you try removing the `node_modules` and `bun.lockb` and try again yet?

Yes

**Comment by PackHubTH** [comment]
Can you try setting cookie value first before other attributes, it might not have been created yet when you set expires

**Comment by topether21** [comment]
Thanks, @PackHubTH l. Let me improve the example. It sets the expiration, and the browser reads the value, but the service does not:

```
import { Elysia } from "elysia";
import { isHtml } from '@elysiajs/html'

const app = new Elysia()

app.get('/', () => '<h1>Hello World</h1>', {
  afterHandle({ response, set, cookie }) {
    console.log("afterHandle", cookie)
    cookie.my_cookie.value = 'my_cookie_value'
    cookie.my_cookie.expires = new Date(Date.now() + 1000 * 60 * 60 * 24)
    console.log("cookie value", cookie.my_cookie.value)
    console.log("cookie expires", cookie.my_cookie.expires)
    if (isHtml(response))
      set.headers['content-type'] = 'text/html; charset=utf8'
  }
})

app.get('/a', ({ cookie: { my_cookie } }) => {
  console.log("cookie value", my_cookie)
  console.log("cookie expires", my_cookie.expires)
  console.log("cookie domain", my_cookie.domain)
  return {
    message: { my_cookie }
  }
})
app.listen(3000);

console.log(
  `🦊 Elysia is running at http://${app.server?.hostname}:${app.server?.port}`
);
```

![Monosnap localhost:3000:a 2024-09-13 10-56-40](https://github.com/user-attachments/assets/666b6c39-c4a2-4b47-b491-ff06b53a737a)

**Comment by SaltyAom** [comment]
Fixed and published under [1.1.13](https://github.com/elysiajs/elysia/releases/tag/1.1.13) with test case included
see https://github.com/elysiajs/elysia/compare/1.1.12...1.1.13

If this doesn't fix the issue, feel free to tag me or reopen the issue.
